test/com/dataengi/crm/identities/daos/UsersDAOSpec.scala

```diff
@@ -54,12 +91,128 @@ class UsersDAOSpec
+      val newCompanyResult: XorType[Long] = companiesService.create(NewCompanyName).await()
+      val newCompany                      = companiesService.get(newCompanyResult.value).await().value
+      val role                            = rolesService.find(RolesConfiguration.CompanyManager.name).await().value
+
+      newCompany.id must not be None
+
+      val companyId      = newCompany.id.get
+      val findUserResult = usersDAO.findByCompany(companyId).await()
+      println(s"[user-dao][find] ${findUserResult.logResult}")
+      findUserResult.isRight === true
+
+      val users = findUserResult.value
+
+      users must have size 0
+
+    }
+
     "all user" in {
       val allResult = usersDAO.all.await()
       println(s"[user-dao][all] ${allResult.logResult}")
       allResult.isRight === true
     }
 
+    "update user" in {
+      val NewCompanyName: String          = "NEW_COMPANY" + Gen.alphaStr.sample.get
+      val newCompanyResult: XorType[Long] = companiesService.create(NewCompanyName).await()
+      val newCompany                      = companiesService.get(newCompanyResult.value).await().value
+      val role                            = rolesService.find(RolesConfiguration.CompanyManager.name).await().value
+
+      val TestLoginInfo3: LoginInfo = LoginInfo("provider3", "dasd3@asd.com")
+      val TestLoginInfo4: LoginInfo = LoginInfo("provider4", "dasd3@asd.com")
+
+      val user3 = User(
+        loginInfo = TestLoginInfo3,
+        company = newCompany,
+        role = role
+      )
+
+      val getUserResult = usersDAO.add(user3).await()
+      println(s"[user-dao][get] ${getUserResult.logResult}")
+      getUserResult.isRight === true
+
+      val key = getUserResult.right.get
+
+      val updateUser       = User(loginInfo = TestLoginInfo4, company = newCompany, role = role).copy(id = Some(key))
+      val updateUserResult = usersDAO.update(updateUser).await()
+      updateUserResult.isRight === true
+
+      val getUser = usersDAO.get(key).await()
+      getUser.isRight === true
+      getUser.value.loginInfo.providerKey === TestLoginInfo4.providerKey
+
+    }
+
+    "update user negative" in {
+      val NewCompanyName: String          = "NEW_COMPANY" + Gen.alphaStr.sample.get
+      val newCompanyResult: XorType[Long] = companiesService.create(NewCompanyName).await()
+      val newCompany                      = companiesService.get(newCompanyResult.value).await().value
+      val role                            = rolesService.find(RolesConfiguration.CompanyManager.name).await().value
+
+      val TestLoginInfo5: LoginInfo = LoginInfo("provider4", "dasd4@asd.com")
+
+      val updateUser       = User(loginInfo = TestLoginInfo5, company = newCompany, role = role).copy(id = Some(56667))
+      val updateUserResult = usersDAO.update(updateUser).await()
+      updateUserResult.isLeft === true
+
+    }
+
+    "getOption user" in {
+      val NewCompanyName: String          = "NEW_COMPANY" + Gen.alphaStr.sample.get
```

Comment:
Prefer name `newCompanyName`

https://github.com/awesome-it-ternopil/best-practices/wiki/Scala-best-practice#naming-convention